pkg/asset/installconfig/vsphere/vsphere.go

```diff
@@ -303,9 +305,12 @@ func getNetwork(ctx context.Context, path string, finder Finder, client *vim25.C
 		return "", errors.New("did not find any networks")
 	}
 	if len(networks) == 1 {
-		n := networks[0].(networkNamer)
-		logrus.Infof("Defaulting to only available network: %s", n.Name())
-		return n.Name(), nil
+		n, err := vspheretypes.GetNetworkName(ctx, client, networks[0])
```

Comment:
Good question :) This change was needed since the networks slice now contains generic `ManagedObjectReference` and no longer the `NetworkReference` that contains `Name()`. AFAIK, we can't easily get a `NetworkReference` out of a `ManagedObjectReference` without going through the network factory used in `GetNetworkName`.